debugger/src/debug_adapter/dap_adapter.rs

```diff
@@ -1217,18 +1284,25 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
         // The MS DAP spec requires that we always have to return a fixed number of instructions.
         while assembly_lines.len() < instruction_count as usize {
             if read_more_bytes {
-                match core_data.target_core.read_word_32(read_pointer) {
+                match target_core.core.read_word_32(read_pointer) {
                     Ok(new_word) => {
                         // Advance the read pointer for next time we need it.
-                        read_pointer += 4;
+                        read_pointer = if let Some(valid_read_pointer) = read_pointer.checked_add(4)
```

Comment:
Please use a proper Option instead of maxing the value.

Reply:
Done :)